contracts/v0.7/bridge/token/optimism/LinkTokenOptimism.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.6.0 <0.8.0;
+
+/* Interface Imports */
+import { IERC165 } from "../../../../../vendor/OpenZeppelin/openzeppelin-contracts/contracts/introspection/IERC165.sol";
+import { ITypeAndVersion } from "../../../../v0.6/ITypeAndVersion.sol";
+import { IERC20Optimism } from "./IERC20Optimism.sol";
+
+/* Contract Imports */
+import { ERC20 } from "../../../../../vendor/OpenZeppelin/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import { LinkToken } from "../../../../v0.6/LinkToken.sol";
+
+/// @dev Access controlled mintable & burnable LinkToken, for use on Optimism L2 network.
+contract LinkTokenOptimism is ITypeAndVersion, IERC20Optimism, LinkToken {
```

Comment:
It's easier if you use 
`import { L2StandardERC20 } from "@eth-optimism/contracts/libraries/standards/L2StandardERC20.sol";` 

and use that as a base.

Reply:
Also `LinkToken` mints the total supply upon creation which we cannot do in L2. Otherwise you'll have 10**27 tokens on L2 that are non withdraw-able.

Reply:
Thx, @elenadimitrova!

We have an internal policy to avoid a public NPM registry, so we lower the risk of supply chain attacks. We currently use submodules (looking to migrate to private NPM registry), but I didn't want to include the whole submodule for that one file so I just c/p it.

There is an `_onCreate` hook in `LinkToken` that mints the supply, but here we override it to avoid initial mint:
```sollidity
/**
   * @dev Overrides parent contract so no tokens are minted on deployment.
   * @inheritdoc LinkToken
   */
  function _onCreate()
    internal
    override
  {}
```